sku.0/sys.server/compiled/game/script/systems/combat/combat_actions.java

```diff
@@ -10483,21 +10478,17 @@ public int bm_revive_pet_1(obj_id self, obj_id target, String params, float defa
             setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
             return SCRIPT_OVERRIDE;
         }
-        if (getDistance(getLocation(self), getLocation(beast)) > 7.0f)
-        {
-            sendSystemMessage(self, new string_id("spam", "stand_next_to_corpse"));
-            setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
-            return SCRIPT_OVERRIDE;
-        }
         if (!combatStandardAction("bm_revive_pet_1", self, target, params, "", ""))
         {
             setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
             return SCRIPT_OVERRIDE;
         }
+        float playerToPetRange = getDistance(getLocation(self), getLocation(beast));
+        float percentRangeMod = (playerToPetRange / 84f) * 100f;
         messageTo(self, "channelRevivePet", null, 0, false);
         float baseCooldownTime = getBaseCooldownTime("bm_revive_pet_1");
         float cooldownTimeMod = getEnhancedSkillStatisticModifierUncapped(self, "expertise_bm_pet_revive_time");
-        setCommandTimerValue(self, TIMER_COOLDOWN, baseCooldownTime - cooldownTimeMod);
+        setCommandTimerValue(self, TIMER_COOLDOWN, (baseCooldownTime - cooldownTimeMod) * percentRangeMod);

```

Comment:
The multiplier should only apply if we are > 7m away? I'm interpreting the patch note to mean that the range was increased to 85, and that there is now a scale beyond the original, not that the logic for the distance from 0m-7m was also adjusted. 

How sure are we on this math to calculate how much the time is extended by with correlation to the range? Is it documented somewhere?

Reply:
Added a if statement to the `percentRangeMod ` float to either be the modified value or 0 depending on the range.
Also added a new dictionary and added a float to pass through to the `channeRevivePet` message and then picked that up in the method to times the duration the same as we do here which is now if you are 84m away you will have a 10x longer cooldown